### ISSUE TITLE
Input with items already selected reads blank when focused with NVDA 

### ISSUE THREAD

**Issue by stephenbe** [select2/select2#4418]
## Prerequisites
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of Select2
- [x] This is not a usage question (Those should be directed to the [community](https://select2.github.io/community.html))
- [x] I have attempted to find the simplest possible steos to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)
## Steps to reproduce the issue
1. Go to this codepen : http://codepen.io/stephendesjardins/pen/jrWmgq
2. NVDA with firefox
3. Select a few options
4. NVDA will not announce what options are already selected
## Expected behavior and actual behavior

I was expecting that NVDA tell us what option are already selected
## Environment

Browsers
- [ ] Google Chrome
- [x] Mozilla Firefox
- [ ] Internet Explorer

Operating System
- [x] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries
- jQuery version: 2.2.4
- Select2 version:4.0.3 and version that is on master
## Isolating the problem
- [x] This bug happens [on the examples page](https://select2.github.io/examples.html)
- [x] The bug happens consistently across all tested browsers
- [x] This bug happens when using Select2 without other pluigns
- [x] I can reproduce this bug in [a codepen](http://codepen.io/stephendesjardins/pen/jrWmgq)

Here is what I mean, when it says "vide", it means "empty"
![select2](https://cloud.githubusercontent.com/assets/3018441/16017759/552519e8-316f-11e6-91db-bc07ac86f32e.gif)

**Comment by kevin-brown** [comment]
This sounds like a duplicate of https://github.com/select2/select2/issues/3735.

**Comment by stephenbe** [comment]
That issue is about changing options in the select-container. I'm talking about when you selected a few options and then you go back to that input, you can't know there are already a few options (see gif attached).

**Comment by ashokramadoss** [comment]
if any fix available fro JAWS16 and NVDA

**Comment by acconrad** [comment]
This seems to be fixed in Firefox and I think this can be closed:

![screen shot 2017-12-13 at 13 54 53](https://user-images.githubusercontent.com/422184/33956661-253fba9e-e037-11e7-9e81-2b573a8ffcac.png)

**Comment by alexweissman** [comment]
@acconrad it appears that this issue is about the NV Access (NVDA) screen reader - are you using that?

**Comment by acconrad** [comment]
@alexweissman you're totally right, I read that wrong. Please ignore, sorry!

**Comment by alexweissman** [comment]
Actually, this is a duplicate of #3735, so we can close it after all!

**Comment by stephenbe** [comment]
@alexweissman I already explained to @kevin-brown that this is another issue and he reopened it 
please see : https://github.com/select2/select2/issues/4418#event-690850959

Do not close this issue. **It is not a duplicate** 
https://github.com/select2/select2/issues/4418#issuecomment-225668558

**Comment by alexweissman** [comment]
> JAWS is silent when changing options, and NVDA reads "blank".

Isn't that the exact same issue you described?

> NVDA will not announce what options are already selected

**Comment by stephenbe** [comment]
@alexweissman 

My issue is when you go back into an input with a few items selected, it reads the input as blank instead of that there is something already in the input. So when someone comes into the input with a few items and it reads blank, it gives the impression that there is nothing there. So the user with a disability will submit a form thinking he has only one item in the input when it has three.

The issue that you are referencing is trying to navigate in the options which read blank.

I would highly recommend you look at the gif I posted in this comment : https://github.com/select2/select2/issues/4418#issue-160010005

You will have a better idea of the difference.

Hopefully you will re-open this issue. If you need any clarification, please let me know.

**Comment by alexweissman** [comment]
Ok, please edit the title of this issue to make the difference more clear, so it doesn't get closed again in the future.

**Comment by stephenbe** [comment]
Changed the title to reflect better the issue. Let me know what you think! thanks

**Comment by alexweissman** [comment]
Thanks.  It could still be the same root cause as #3735, but we'll leave it open until you have time to investigate the cause of these issues more 😄

**Comment by Amarjit** [comment]
An interim fix would be to disable the auto-complete.

```
    $("#your_select_element").on('select2:open', function (e) {
        $('.select2-search input').prop('focus',false);
    });
```

**Comment by stephenbe** [comment]
please don't close this issue. It is very much still a problem

**Comment by kevin-brown** [comment]
Reopening this since it's still not fixed and we still consider it to be an issue.

We welcome pull requests that fix this issue, by the way.

**Comment by kevin-brown** [comment]
Still welcoming pull requests, I'll gladly review them.

**Comment by stephenbe** [comment]
Hi @kevin-brown ,

I made a PR that fixes this issue in firefox. Let me know if you have any feedback on it.

Let's get this done!

**Comment by kevin-brown** [comment]
I have opened a pull request at https://github.com/select2/select2/pull/5842 which aims to redo the accessibility of the selection area with multiple and single selects to make them more accessible to screen readers. One of the main changes involves changing the selection to ensure the selected items are read out to the user when the search and other areas are focused.

While I understand that it has been a while, it would be greatly appreciated if anyone could help us test out these changes to ensure this ticket has been fixed. You can test these on your own by recompiling Select2 from the lastest `develop` branch, or you can wait for the next beta release (which should be out once that pull request is merged). Our goal is to significantly improve the accessibility in time for the upcoming 4.1.0 release of Select2.
